File: Swift/prime-number-of-set-bits-in-binary-representation.swift
class Solution {
    func countPrimeSetBits(_ L: Int, _ R: Int) -> Int {
        var res = 0
        let primes = Set([2, 3, 5, 7, 11, 13, 17, 19])

        for i in L...R {
            res += primes.contains(bitCount(i)) ? 1 : 0
        }

        return res
    }

    private func bitCount(_ n: Int) -> Int {
        var count = 0
        var n = n

        while n > 0 {
            n &= n - 1
            count += 1
        }

        return count
    }
}

File: Python/prime-number-of-set-bits-in-binary-representation.py
class Solution:
    def countPrimeSetBits(self, L, R):
        """
        :type L: int
        :type R: int
        :rtype: int
        """
        primes = set([2, 3, 5, 7, 11, 13, 17, 19])
        return sum(self.bitCount(i) in primes
                    for i in range(L, R + 1))
        
    def bitCount(self, n):
        count = 0

        while n:
            n &= n - 1
            count += 1

        return count
